src/ConstructiveSolidGeometry/IO.jl

```diff
@@ -11,26 +11,35 @@ const CSG_dict = Dict{String, Any}(
     # "union" => CSGUnion,
     # "difference" => CSGDifference,
     # "intersection" => CSGIntersection,
-    "translate" => CartesianVector,
-    "rotate" => Rotation, # type from Rotations.jl: Rotation.Rotation
-    # "scale" => ScaledGeometry
 )
 
 function get_geometry_key(::Type{T}, dict::AbstractDict, input_units::NamedTuple) where {T}
-    dict_keys = filter(k -> k in keys(CSG_dict), keys(dict))
-    transformations = sort!(filter(k -> 
-                                (k == "translate" && !any(broadcast(key -> key in keys(CSG_dict), keys(dict["translate"])))) || 
-                                (k == "rotate" && !any(broadcast(key -> key in keys(CSG_dict), keys(dict["rotate"])))), 
-                            collect(dict_keys)), 
-                      rev = true) #this will ensure that the rotation is parsed first and the translation is parsed second
-    primitives = setdiff(dict_keys, transformations)
-    @assert length(primitives) <= 1 "Too many geometry entries in dictionary: $(length(dict_keys))."
-    @assert length(primitives) >= 1 "None of the entries $(keys(dict)) describes a Geometry."
-    transformations = broadcast(t -> parse_CSG_transformation(T, dict, CSG_dict[t], input_units), transformations)
-    first(primitives), isempty(transformations) ? missing : transformations
+    g = collect(filter(k -> k in keys(CSG_dict), keys(dict)))
+    @assert !(length(g) > 1) "Too many geometry entries in dictionary: $(length(g))."
+    @assert !(length(g) == 0) "None of the entries $(keys(dict)) describes a Geometry."
+    g[1]
 end
 
 
+function parse_CSG_transformation(::Type{T}, dict::AbstractDict, input_units::NamedTuple) where {T}
+    r = if haskey(dict, "rotation")
+        parse_rotation_vector(T, dict["rotation"], input_units.length)
```

Comment:
This used to be `"rotate"`. Did it change to `"rotation"`?
I find this somewhat inconsistent with respect to `"scale"` and `"translate"`

Reply:
No, thanks. I will change it.

Reply:
Also, it says `parse_rotation_vector`. It should be `parse_rotation_matrix`

Reply:
And input units should be `input_units.angle`

Reply:
Ah, right. I changed it.